pkg/agent/util/syscall/syscall_windows.go

```diff
@@ -0,0 +1,416 @@
+	IdxSetIPInterface       = iota
+	IdxCreateIPForwardEntry = iota
+	IdxDeleteIPForwardEntry = iota
+)
+
+type NetIOInterface interface {
+	GetIPInterfaceEntry(ipInterfaceRow *MibIPInterfaceRow) (errcode error)
+
+	SetIPInterfaceEntry(ipInterfaceRow *MibIPInterfaceRow) (errcode error)
+
+	CreateIPForwardEntry(ipForwardEntry *MibIPForwardRow) (errcode error)
+
+	DeleteIPForwardEntry(ipForwardEntry *MibIPForwardRow) (errcode error)
+
+	ListIPForwardRows(family uint16) ([]MibIPForwardRow, error)
+}
+
+type NetIO struct {
+	syscallN func(trap uintptr, args ...uintptr) (r1, r2 uintptr, err syscall.Errno)
+}
+
+func NewNetIO(syscallNFunc func(trap uintptr, args ...uintptr) (r1, r2 uintptr, err syscall.Errno)) *NetIO {
+	return &NetIO{syscallN: syscallNFunc}
+}
+
+func (n *NetIO) GetIPInterfaceEntry(ipInterfaceRow *MibIPInterfaceRow) (errcode error) {
+	r0, _, _ := n.syscallN(procGetIPInterfaceEntry.Addr(), uintptr(unsafe.Pointer(ipInterfaceRow)))
+	if r0 != 0 {
+		errcode = syscall.Errno(r0)
+	}
+	return
+}
+
+func (n *NetIO) SetIPInterfaceEntry(ipInterfaceRow *MibIPInterfaceRow) (errcode error) {
+	r0, _, _ := n.syscallN(procSetIPInterfaceEntry.Addr(), uintptr(unsafe.Pointer(ipInterfaceRow)))
+	if r0 != 0 {
+		errcode = syscall.Errno(r0)
+	}
+	return
+}
+
+func (n *NetIO) CreateIPForwardEntry(ipForwardEntry *MibIPForwardRow) (errcode error) {
+	r0, _, _ := n.syscallN(procCreateIPForwardEntry.Addr(), uintptr(unsafe.Pointer(ipForwardEntry)))
+	if r0 != 0 {
+		errcode = syscall.Errno(r0)
+	}
+	return
+}
+
+func (n *NetIO) DeleteIPForwardEntry(ipForwardEntry *MibIPForwardRow) (errcode error) {
+	r0, _, _ := n.syscallN(procDeleteIPForwardEntry.Addr(), uintptr(unsafe.Pointer(ipForwardEntry)))
+	if r0 != 0 {
+		errcode = syscall.Errno(r0)
+	}
+	return
+}
+
+func (n *NetIO) freeMibTable(table unsafe.Pointer) {
+	n.syscallN(procFreeMibTable.Addr(), uintptr(table))
+	return
+}
+
+func (n *NetIO) getIPForwardTable(family uint16, ipForwardTable **MibIPForwardTable) (errcode error) {
+	r0, _, _ := n.syscallN(procGetIPForwardTable.Addr(), uintptr(family), uintptr(unsafe.Pointer(ipForwardTable)))
+	if r0 != 0 {
+		errcode = syscall.Errno(r0)
+	}
+	return
+}
+
+func (n *NetIO) ListIPForwardRows(family uint16) ([]MibIPForwardRow, error) {
+	var table *MibIPForwardTable
+	err := n.getIPForwardTable(family, &table)
+	if table != nil {
+		defer n.freeMibTable(unsafe.Pointer(table))
+	}
+	if err != nil {
+		return nil, os.NewSyscallError("iphlpapi.GetIpForwardTable", err)
+	}
+	return unsafe.Slice(&table.Table[0], table.NumEntries), nil
+}
+
+func NewIPForwardRow() *MibIPForwardRow {
+	return &MibIPForwardRow{
+		SitePrefixLength:     255,
+		Metric:               0,
+		Loopback:             true,
+		AutoconfigureAddress: true,
+		Publish:              true,
+		Immortal:             true,
+		ValidLifetime:        0xffffffff,
+		PreferredLifetime:    0xffffffff,
+		Protocol:             RouteProtocolOther,
+	}
+}
+
+func GetNetIoIndex(trap uintptr) int {
```

Comment:
Is this for testing only? If yes, we can move test code out of the file using the other way I suggested in other place

Reply:
It's for testing only, considering the previous [discussion](https://github.com/antrea-io/antrea/pull/4898#discussion_r1274056684), not sure if we need testing package anymore, I'll put it in `_test.go`.